release-notes/24.05/index.html

```diff
@@ -0,0 +1,150 @@
+                <div class="dropdown-menu dropdown-menu-end">
+                  <a class="dropdown-item" href="https://www.cl.cam.ac.uk/research/security/ctsrd/cheri/cheri-slack.html"><i class="bi bi-slack me-2"></i>Talk to Us on Slack</a>
+                  <a class="dropdown-item" href="https://lists.cam.ac.uk/sympa/info/cl-cheribsd-announce"><i class="bi bi-megaphone me-2"></i>Sign Up for Email Updates</a>
+                  <a class="dropdown-item" href="https://lists.cam.ac.uk/sympa/info/cl-cheribsd-discuss"><i class="bi bi-chat-right-text me-2"></i>Join Our Email Discussion List</a>
+                </div>
+              </li>
+            </ul>
+          </div>
+        </div>
+      </div>
+    </header>
+
+    <main class="flex-grow-1">
+      <section class="container py-3">
+        <div class="bg-light border rounded-3 p-4">
+          <h1>CheriBSD 24.05 Release Notes</h1>
+          <h2>Overview</h2>
+          <p>
+          Welcome to the Spring 2024 (24.05) release of the CheriBSD operating system.  As well as a general update of the baseline FreeBSD OS (corresponding to CURRENT as of January 2024) from which CheriBSD is derived, we have introduced several new research components which will be of interest to the CHERI and CheriBSD community:
+          </p>
+          <ul>
+            <li>
+            Library Compartmentalization (<a href="https://man.cheribsd.org/cgi-bin/man.cgi/release-24.05/c18n">https://man.cheribsd.org/cgi-bin/man.cgi/release-24.05/c18n</a>) is increasingly mature.  Virtually all software can now run compartmentalized.  It can be enabled system wide by setting the <code>security.cheri.lib_based_c18n_default</code> sysctl to <code>1</code>.  In addition to stronger compatability, a new <code>procstat</code> command <code>c18n</code> provides additional monitoring.  While this implementation is now capabilty of running the full desktop environment discussed below, it is neither complete nor intended to be secure.
+            </li>
+            <li>
+            To strengthen library compartmentalization, we have completed an adjustment to the ABI for functions whose arguments do not fit in registers.  These arguments are now reachable via a bounded capability passed in register C9 and newly compiled callees access them via C9 allowing callers to place argument values anywhere they desire.  This is compatible with code compiled for 23.11, which used a transitional ABI which set C9 in the caller while the callee continued to access variables directly on the stack.  Code compiled for releases prior to 23.11 is no longer supported.
+            </li>
+            <li>
+            We have swiched the default kernel configuration to <code>GENERIC-MORELLO-PURECAP</code>, a pure-capability kernel.  This means that the entire base OS stack is running with spatial memory safety.  Performance of the pure-capability kernel is less well studied than the hybrid kernel so care should be taken when reporting results.
```

Comment:
Not that it hugely matters, but has anyone tested NFS root on the new kernels?

Reply:
I don't think anyone has, but, if it's important, please open an issue on the main cheribsd repo so we can track that.